src/server/render.js

```diff
@@ -6,20 +6,13 @@ import Router from 'react-router';
 import config from './config';
 import initialState from './initialstate';
 import routes from '../client/routes';
+import extend from 'extend';
 import {state} from '../client/state';
 
-export default function render(req, res, locale) {
+export default function render(req, res, preloadedState) {
   const url = req.originalUrl;
-  return loadData(url, locale)
-    .then((appState) => renderPage(res, appState, url));
-}
-
-function loadData(url, locale) {
```

Comment:
I think we should keep single responsibility principles here. Render is just a render. All other things, like fetching additional state should be done in a route handler. This will allow better modifications and extendability in the future, when a new folder e.g called `handlers` lands.